src/services/api/admin/donate/bank-details-foreign/bank-details-foreign-api.ts

```diff
@@ -0,0 +1,36 @@
+import { AxiosInstance } from 'axios';
+import { API_ROUTES } from '../../../../../const/common/api-routes/main-api';
+import { ForeignBankDetailsType, BankCurrency } from '../../../../../types/admin/donate';
+
+export const ForeignBankDetailsApi = {
+    getAll: async (client: AxiosInstance, currency: BankCurrency): Promise<ForeignBankDetailsType[]> => {
+        const response = await client.get<ForeignBankDetailsType[]>(API_ROUTES.DONATE.BANK_DETAILS_FOREIGN, {
+            params: { currency },
+        });
+        return response.data;
+    },
+
+    create: async (
+        client: AxiosInstance,
+        bankDetails: Omit<ForeignBankDetailsType, 'id'>,
+    ): Promise<ForeignBankDetailsType> => {
+        const response = await client.post<ForeignBankDetailsType>(API_ROUTES.DONATE.BANK_DETAILS_FOREIGN, bankDetails);
+        return response.data;
+    },
+
+    update: async (
+        client: AxiosInstance,
+        id: number,
+        bankDetails: Partial<ForeignBankDetailsType>,
+    ): Promise<ForeignBankDetailsType> => {
+        const response = await client.put<ForeignBankDetailsType>(
```

Comment:
Is there a partial or entire update, because I see `Partial<ForeignBankDetailsType>`? If partial use `.patch` instead
```suggestion
        const response = await client.patch<ForeignBankDetailsType>(
```

According to REST conventions, `PUT` is typically used to replace an entire resource, while `PATCH` is the standard method for applying partial updates.

Reply:
I agree with you, but on the backend we only have an endpoint with PUT.